src/litmodels/integrations/checkpoints.py

```diff
@@ -25,6 +28,72 @@
         import pytorch_lightning as pl
 
 
+# Create a singleton upload manager
+@lru_cache(maxsize=None)
+def get_model_manager():
+    """Get or create the singleton upload manager."""
+    return ModelManager()
+
+
+class ModelManager:
+    """Manages uploads and removals with a single queue but separate counters."""
+
+    def __init__(self):
+        """Initialize the ModelManager with a task queue and counters."""
+        self.task_queue = queue.Queue()
+        self.upload_count = 0
+        self.remove_count = 0
+        self._worker = threading.Thread(target=self._worker_loop, daemon=True)
```

Comment:
How does it behaves with keyboard interruption ?

Reply:
this will likely continue

Reply:
Yes, it should. But worth to see how it behaves in general. I think W&B still allows you to stop if you do multiple keyboard interrupt